gufe/tests/test_smallmoleculecomponent.py

```diff
@@ -339,6 +339,35 @@ def test_to_dict(self, phenol):
 
         assert isinstance(d, dict)
 
+    def test_to_dict_hybridization(self, phenol):
+        """
+        Make sure dict round trip saves the hybridization
+        <https://github.com/OpenFreeEnergy/gufe/issues/407>
+        """
+        phenol_dict = phenol.to_dict()
+        TOKENIZABLE_REGISTRY.clear()
+        new_phenol = SmallMoleculeComponent.from_dict(phenol_dict)
+        for atom in new_phenol.to_rdkit().GetAtoms():
+            if atom.GetAtomicNum() == 6:
+                assert atom.GetHybridization() == Chem.rdchem.HybridizationType.SP2
+
+    def test_from_dict_missing_hybridization(self, phenol):
+        """
+        For backwards compatibility make sure we can create an SMC with missing hybridization info.
+        """
+        phenol_dict = phenol.to_dict()
+        new_atoms = []
+        for atom in phenol_dict["atoms"]:
+            # remove the hybridization atomic info which should be at index 7
+            new_atoms.append(tuple([atom_info for i, atom_info in enumerate(atom) if i != 7]))
+        phenol_dict["atoms"] = new_atoms
+        new_phenol = SmallMoleculeComponent.from_dict(phenol_dict)
+        # they should be different objects due to the missing hybridization info
+        assert new_phenol != phenol
+        # make sure the rdkit objects are different
+        for atom_hybrid, atom_no_hybrid in zip(phenol.to_rdkit().GetAtoms(), new_phenol.to_rdkit().GetAtoms()):
+            assert atom_hybrid.GetHybridization() != atom_no_hybrid.GetHybridization()
+
```

Comment:
@IAlibay this test covers loading an old SMC with the missing info it looks like Rdkit defaults to setting the hybridization as undefined.

Reply:
Ok thanks - in that case we should let users know they've hit a bug (just so that they don't end up silently dealing with this). Could you add a warning please?

Reply:
Added the warning though this might become a bit much as every SMC loaded from file created with gufe<1.1.0 will trigger this.